### PR TITLE
Set up sphinx-apidoc to generate API docs

### PR DESCRIPTION
**Pull request by underyx**
### Check list

- [x] I have rebased properly.
      [Please see this for a tutorial.](http://coala.readthedocs.io/en/latest/Developers/Git_Basics.html#rebasing)
- [x] I have gone through the
      [commit guidelines](http://coala.readthedocs.io/en/latest/Developers/Writing_Good_Commits.html)
      and I've followed them:
- [x] I have followed the
       [guidelines for the commit shortlog](http://docs.coala.io/en/latest/Developers/Writing_Good_Commits.html#shortlog).
- [x] I have followed the
       [guidelines for the commit body](http://docs.coala.io/en/latest/Developers/Writing_Good_Commits.html#commit-body).
- [ ] I have included the issue URL with the  `Fixes` keyword if the
       commit fixes a *real bug* and the `Closes` keyword if it adds a
       feature/enhancement.
       [See more here.](http://docs.coala.io/en/latest/Developers/Writing_Good_Commits.html#issue-reference)
- [ ] I have [run all the tests](http://coala.readthedocs.io/en/latest/Developers/Executing_Tests.html)
      and they all pass. If any CI (below) is not green, please fix them. If
      GitMate issues appear you will have to amend your commits, pushing new
      commits on top is not sufficient!

### Reviewers

@sils @Adrianzatreanu

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -1,5 +1,14 @@
+coala API documentation
+=======================
+
+Hey there! This is the documentation for developers of the coala project
+itself. If you're trying to **use** coala, you should doc have a look at our
```
Comment (Adrianzatreanu) [inline docs/index.rst]: "you should doc have a look" makes no sense to me ;d phrased it badly?

**Comment by Adrianzatreanu** [comment]
It also seems that circleCI is failing due to you creating some moduels which arent included in the toctree (index.rst)

File: docs/index.rst (hunk 1 above)
Comment (underyx) [inline docs/index.rst]: Doc shouldn't be there; accidentally left it after a rewrite.

**Comment by underyx** [comment]
@Adrianzatreanu yup, apparently `sphinx-apidoc` automatically creates a table of contents file (basically the same as the `index.rst` I added). I've added the `--no-toc` flag to disable this behavior.

**Comment by sils** [comment]
ack https://github.com/coala/coala/pull/2969/commits/b3f66025288ba4b27bdd665febabb8f56b593757

**Comment by sils** [comment]
@rultor merge

**Pull request merged**